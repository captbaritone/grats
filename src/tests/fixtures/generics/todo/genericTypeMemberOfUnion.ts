/** @gqlType */
export class User<T> {
  __typename: "User";
  /** @gqlField */
  name: string;

  /** @gqlField */
  friend: T;
}

/** @gqUnion */
type Friendly = User<Dog>;

/** @gqlType */
class Dog {
  /** @gqlField */
  name: string;
}
