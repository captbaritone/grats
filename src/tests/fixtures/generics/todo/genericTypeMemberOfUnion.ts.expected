-----------------
INPUT
----------------- 
/** @gqlType */
export class User<T> {
  __typename: "User";
  /** @gqlField */
  name: string;

  /** @gqlField */
  friend: T;
}

/** @gqUnion */
type Friendly = User<Dog>;

/** @gqlType */
class Dog {
  /** @gqlField */
  name: string;
}

-----------------
OUTPUT
-----------------
-- SDL --
type Dog {
  name: String @metadata
}
-- TypeScript --
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const DogType: GraphQLObjectType = new GraphQLObjectType({
        name: "Dog",
        fields() {
            return {
                name: {
                    name: "name",
                    type: GraphQLString
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [DogType]
    });
}
